server/controllers/finance/cost_center_allocation_bases.js

```diff
@@ -0,0 +1,46 @@
+/**
+* Cost Center Allocation Keys
+*
+* This controller exposes an API to the client for reading allocation keys
+*/
+const _ = require('lodash');
+const db = require('../../lib/db');
+
+async function fetch() {
+  const queryCostCenterIndexesList = `
+    SELECT 
+      fc.id, 
+      ccb.name AS cost_center_allocation_basis_label, 
+      ccbv.quantity, fc.label AS cost_center_label,
+      fc.step_order 
+    FROM cost_center fc 
+    JOIN cost_center_allocation_basis_value ccbv ON ccbv.cost_center_id = fc.id 
+    JOIN cost_center_allocation_basis ccb ON ccb.id = ccbv.basis_id
+    ORDER BY fc.step_order ASC;
+  `;
+  const costCenterIndexesList = await db.exec(queryCostCenterIndexesList);
+  const indexes = _.groupBy(costCenterIndexesList, 'cost_center_allocation_basis_label');
+  const costCenterList = [];
+  const costCenterIndexes = _.keys(indexes).map((index, i) => {
+    const fcIndex = _.sortBy(indexes[index], 'step_order');
```

Comment:
Are they not already ordered by step order?  I thought the SQL would take care of that.

Reply:
In the SQL procedure it was not obvious to sort these data.